### ISSUE TITLE
WandB not logging the default metrics properly (axis scales)

### ISSUE THREAD

**Issue by Petros626** [ultralytics/ultralytics#18844]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello guys,

I wonder why the standard metrics are not logged related to the steps. The terminal output shows e.g. 4/120, but the point in the web interface is in the middle.

Log in Terminal:
```txt

default log before....

class BaseTrainer: used validator <ultralytics.models.yolo.obb.val.OBBValidator object at 0x7f901022f280>
class BaseTrainer: used metrics ['metrics/precision(B)', 'metrics/recall(B)', 'metrics/mAP50(B)', 'metrics/mAP50-95(B)', 'val/box_loss', 'val/cls_loss', 'val/dfl_loss']
[34m[1moptimizer:[0m AdamW(lr=0.0026245248437283047, momentum=0.837) with parameter groups 63 weight(decay=0.0), 73 weight(decay=0.0006), 72 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to [1mkitti_bev_yolo_wandb/epo._120_bat._4[0m
Starting training for 120 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/120      1.85G      1.064      1.246      1.989         50        640:  89%|████████▉ | 4139/4635 [08:15<00:57,  8.61it/s] 

pass some time

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
            4/120      1.84G     0.5208      0.457      1.838         11        640:   0%|          | 1/4635 [00:00<08:52,  8.71it/s]
```

But the graph stays in the same position:
![Image](https://github.com/user-attachments/assets/2246a49f-a904-4392-9025-acd4789b330d)

How frequently the metrics are updated? Can I change this frequency to see faster results?
Thanks for any suggestion!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Petros626, thank you for your interest in Ultralytics 🚀!

We recommend reviewing the [Ultralytics Docs](https://docs.ultralytics.com/) for helpful guidance. For example, our [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage tutorials often address common issues.

It appears you're experiencing an issue with WandB logging metrics during training 🐛. To help us investigate, could you please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) that focuses solely on the bug? This will help pinpoint whether the issue stems from your code, the integration with WandB, or elsewhere in your training workflow.

To ensure everything is up-to-date, please upgrade to the latest `ultralytics` package and validate your issue is not already resolved:

```bash
pip install -U ultralytics
```

This includes verifying compatibility with the [required dependencies](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Ensure you are working in a **Python>=3.8** environment with **PyTorch>=1.8**.

## Debugging Environment Check

Additionally, you can validate your environment using any of these YOLO-compatible setups:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- [Google Cloud Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- [AWS Deep Learning AMI Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

---

For real-time conversations with the community, you can join our [Discord](https://discord.com/invite/ultralytics) 🎧. Alternatively, head over to [Discourse](https://community.ultralytics.com/) for in-depth discussions, or engage on our [Subreddit](https://reddit.com/r/Ultralytics) for broader community support.

⚡ This is an automated response. An Ultralytics engineer will follow up with you soon to provide further assistance on your issue. Thank you for your patience!

**Comment by Y-T-G** [comment]
You should follow the guide:

https://docs.ultralytics.com/integrations/weights-biases/

It doesn't add callbacks.

**Comment by Petros626** [comment]
Thanks for referring the duplicate. I'll give this a try.

**Comment by Y-T-G** [comment]
That was the old method. You don't need to do that now.

**Comment by Y-T-G** [comment]
WandB docs is outdated too

**Comment by Petros626** [comment]
And where is a updated guide to use `wandb` with sweeps?

**Comment by Petros626** [comment]
> > So is your suggestion for training without sweeps?
> 
> I am not sure what you mean here. The reason you don't need to perform `init()` is because Ultralytics does that automatically.
> 
> [ultralytics/ultralytics/utils/callbacks/wb.py](https://github.com/ultralytics/ultralytics/blob/5d9535b008b92ba86e90b1a67281dbd93f53e43e/ultralytics/utils/callbacks/wb.py#L113)
> 
> Line 113 in [5d9535b](/ultralytics/ultralytics/commit/5d9535b008b92ba86e90b1a67281dbd93f53e43e)
> 
>  wb.init( 
> But if you do that manually, it should handle that and avoid double initialization. Just don't use the `add_wandb_callback` which is outdated and breaks things.

I mean how can I train then a model with sweeps for hyperparameter optimization using the new training pattern, not the old one.

**Comment by Y-T-G** [comment]
Remove `add_wandb_callback`.

You can add back init.

The rest should work fine.

**Comment by Petros626** [comment]
I'll try it, but can't until https://github.com/ultralytics/ultralytics/issues/9856#issuecomment-2612403118 is solved.
